hw/top_earlgrey/dv/chip_sim_cfg.hjson

```diff
@@ -148,11 +148,13 @@
     {
       name: build_seed
       pre_build_cmds: [
-        // TODO: find where in run phase we are using this pkg. It fails during ibex TLUL integrity
-        // check.
-        // '''cd {proj_root} && ./util/topgen.py -t {ral_spec} \
-        //        -o hw/top_earlgrey --rnd_cnst_seed {seed}
-        // ''',
+        // Generate RndCnsts
+        '''cd {proj_root} && ./util/topgen.py -t {ral_spec} \
+               -o hw/top_earlgrey --rnd_cnst_seed {seed}
+        ''',
+        // Generate LC encoding
+        "cd {proj_root} && ./util/design/gen-lc-state-enc.py --seed {seed}",
+        // Generate OTP memory map and scrambling constants keys.
```

Comment:
I have a comment for line 162 below. 
Could we save the `{build_seed_file_path}` to `{scratch_path}` area instead? The `{eval_cmd}` is not expected to cause any side-effects in the execution environment. It was meant to evaluate expressions to set variables to specific values.

Reply:
yeah we can do that - but wouldn't that cause issues with builds that are ongoing in parallel?
the build seed files are currently only uniquified by path and not by name: `"{build_dir}/build_seed.log"`

Reply:
Ok that makes sense. I was referring to separately invoked builds (not builds that are invoked with the same dvsim command), but locking the project root directory should take care of that issue. I can change this to `{scratch_path}/build_seed.log`.

Reply:
Right, but if set the scratch root to a different directory than the proj root, we will need to add a lock to the scratch root as well.

Reply:
Hm, if we `flock`ed the `scratch_path` in the same way, does that really work? The lock is only active for the pre-build commands right? So if we locked the `scratch_path` in the same way, the lock would not persist through the run phase(s) where the seed file is consumed, and hence would not protect multiple separate dvsim invocations from stepping onto each other.

It almost feels like it would (for now) be simpler to leave the build seed file inside the build folder, since that way it is conveniently uniquified via the path...

Reply:
Ever since #16052 was merged, the dvsim `build_seed` is [now available in both the build and run phases](https://github.com/lowRISC/opentitan/pull/16052#discussion_r1015952226), so do we even need the file anymore?

Reply:
I feel like the logic update in dvsim in #16052 needs a bit more srcutiny. You are right @msfschaffner - the file containing the seed value needs to exist in the build directory, so that tests can run in our infra (built database is tarballed, uploaded and added as input to sim runs). So anything that exists outside of the build dir will not be visible to the VMs running the sims.

Reply:
I am not sure why the `eval_cmd` and `mkdir -p {build_dir}` were needed on line 160. If you see `sim.mk`, it creates the build dir before executing the `pre_build_cmds`. So it should be sufficient to just do `echo {seed} > {build_seed_file_path}`.

Reply:
I'm not sure why we need this file at all once we [move to only using Bazel generated OTP images](https://github.com/lowRISC/opentitan/issues/15337#issuecomment-1332758363). dvsim provides the build seed in the templating environment in the sim.mk file, so no need to persist the build seed from build to run phases, as it is available already in both in the templating environment no?

Reply:
i guess you need it until then tho

Reply:
I think we passed this by file before since the `{build_seed}` wildcard is only available when dvsim is invoked with the `--build-seed` switch. Otherwise substitution fails (hence also the inline conditional snippet in the `gen_otp_images_cmd_opts` assignment).

Given that we are going to replace this mechanism soon with the bazel variant, I would just leave it as is for now (and remove it when we switch).

Reply:
Sorry I was in the middle of explaining this when you updated the PR hehe, but my understanding was that during the dvsim build phase the "build seed" is just called `seed`, but in the run phase it is `build_seed`, since after #16052, the run phase gets [two seeds now](https://cs.opensource.google/opentitan/opentitan/+/master:util/dvsim/Deploy.py;drc=8e586da52d5d7bdeb3811d894357b48ad48e21e2;l=472), the seed used during the "build phase", called `build_seed` and the seed used during the run phase called `seed`, hence if you look at the seeds passed to the bazel_opts in the `sw_build` step of the `sim.mk` file (which runs at the "run phase") both `seed` and `build_seed` are used. Does that make sense? Each phase used to just get one seed, while now the "run phase" gets its seed plus the build phase's seed. Hence why the ["gen_otp_images_cmd_opts"](https://cs.opensource.google/opentitan/opentitan/+/master:hw/top_earlgrey/dv/chip_sim_cfg.hjson;drc=809eb00b295f44e7b646f2ca045c3c6776d9191e;l=204) in the chip_sim_cfg.hjson file had to read the build phase seed from a file.

The way you have it looks right now It think.

Reply:
agreed with the leave as is for now until we replace with bazel mechanims comment @msfschaffner